src/main/scala/controllers/ActionHandler.scala

```diff
@@ -16,32 +17,40 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
   import ApplicationData._
   import FieldCheckHelpers._
 
-
   def doSave(id: ApplicationId, sectionNumber: Int, fieldValues: JsObject): Future[Result] = {
-    sectionTypeFor(sectionNumber) match {
-      case VanillaSection =>
-        JsonHelpers.allFieldsEmpty(fieldValues) match {
-          case true => applications.deleteSection(id, sectionNumber).map(_ => redirectToOverview(id))
-          case false => applications.saveSection(id, sectionNumber, fieldValues).map(_ => redirectToOverview(id))
+    gatherSectionDetails(id, sectionNumber).flatMap {
+      case Some(app) =>
+        app.formSection.sectionType match {
+          case SectionTypeForm =>
+            JsonHelpers.allFieldsEmpty(fieldValues) match {
+              case true => applications.deleteSection(id, sectionNumber).map(_ => redirectToOverview(id))
```

Comment:
At the moment that's never the case (since there is always at least one mandatory field in all of our sections). If we did encounter in the future a section where all fields could legitimately be blank then we might address it. But, looking at the code, it would already be possible to mark an empty section as complete if there were no mandatory fields. The section data would be saved with empty values and the completed at date would be set. It's only on the "Save and continue" that the section is deleted on empty. The net effect of that is that it's not possible for a section with all fields empty to be in an In Progress state.